File: Programmers/Python3/lv3/전화번호 목록.py
'''
풀이
길이순으로 정렬 후 이후 문자열을 짧은 길이의 문자열만큼 슬라이싱 해서 같은게 있는지 확인 
'''
phone_book=["119", "97674223", "1195524421"]
def solution(phone_book):
    phone_book.sort(key=lambda x: len(x))
    book_num=len(phone_book)
    for i in range(book_num):
        for j in range(i+1,book_num):
            if phone_book[i]==phone_book[j][:len(phone_book[i])]:
                return False    
    return True

print(solution(phone_book))

File: Programmers/Java11/src/com/programmers/전화번호목록.java
import java.util.ArrayList;
import java.util.Arrays;
import java.util.Collections;
import java.util.List;

public class 전화번호목록 {
    static class Solution {
        public boolean solution(String[] phone_book) {
            for (int i = 0; i < phone_book.length; i++) {
                for (int j = i+1; j < phone_book.length; j++) {
                    if(phone_book[j].startsWith(phone_book[i])) return false;
                    if(phone_book[i].startsWith(phone_book[j])) return false;
                }
            }

            return true;
        }
    }

    public static void main(String[] args) {

        String[] phone_book = {"12","123","1235","567","88"};

        전화번호목록.Solution solution = new 전화번호목록.Solution();
        boolean solution1 = solution.solution(phone_book);
        System.out.println(solution1);

    }
}
